### PR TITLE
[GEOS-10929] isolated workspace may cause broken links between catalo…

### PR DESCRIPTION
**Pull request by NielsCharlier**
[GEOS-10929](https://osgeo-org.atlassian.net/browse/GEOS-10929)

…g objects

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/community/wps-sextante/src/main/java/org/geoserver/wps/sextante/GTVectorLayer.java [hunk 1]
```diff
@@ -26,6 +25,7 @@
 import org.geotools.feature.FeatureCollection;
 import org.geotools.feature.simple.SimpleFeatureBuilder;
 import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.geotools.util.logging.Logging;
```
Comment (aaime) [inline src/community/wps-sextante/src/main/java/org/geoserver/wps/sextante/GTVectorLayer.java:28]: What's happening here? Unrelated reformats here and in a few other files.

File: src/main/src/main/java/org/geoserver/catalog/impl/CatalogImpl.java [hunk 2]
```diff
@@ -141,6 +141,7 @@ public Iterable<CatalogValidator> getValidators() {
     }
 
     public void setFacade(CatalogFacade facade) {
+        rawFacade = facade;
```
Comment (aaime) [inline src/main/src/main/java/org/geoserver/catalog/impl/CatalogImpl.java:144]: So the actual important change is here... the refactor of the event management obfuscates it (still a good thing, but it's a distraction when trying to find what makes the fix actually work).

File: src/main/src/main/java/org/geoserver/catalog/impl/CatalogImpl.java [hunk 3]
```diff
@@ -2277,4 +2246,12 @@ public <T extends CatalogInfo> T get(Class<T> type, Filter filter)
     public CatalogCapabilities getCatalogCapabilities() {
         return facade.getCatalogCapabilities();
     }
+
+    public CatalogImpl getRawCatalog() {
```
Comment (aaime) [inline src/main/src/main/java/org/geoserver/catalog/impl/CatalogImpl.java:2250]: Bit of Javadoc with an explanation of why the original CatalogImpl is not "raw"?

**Review by aaime — CHANGES_REQUESTED** [review]

**Comment by NielsCharlier** [comment]
@aaime thanks for the review, made the changes

**Pull request merged**